### ISSUE TITLE
The AArch64 feature list is incomplete.

### ISSUE THREAD

**Issue by jacobbramley** [rust-lang/stdarch#993]
The set of features in `arch/aarch64.rs` is rather behind the architecture (and what's described by Linux's `auxv`).

I went to bring the list up to date, and noticed that a few are actually listed (in `os/linux/aarch64.rs`), but commented out. Are they deliberately omitted? If so, what criteria are used for inclusion here?

**Comment by Amanieu** [comment]
It's just because these features are not supported [here](https://github.com/rust-lang/rust/blob/master/compiler/rustc_codegen_ssa/src/target_features.rs#L35) in the compiler. You are welcome to update both to add the latest features.

**Comment by jacobbramley** [comment]
Oh, thanks for that pointer! I will do that; it makes sense to keep these in agreement.

Actually _making use_ of them in the compiler will be rather more difficult, of course. Can I assume that all that is required is awareness (e.g. for `asm!`)?

**Comment by Amanieu** [comment]
The main concern is that the compiler and stdarch (and LLVM to some extent) agree on the same name for a particular feature. You should check against the feature names that LLVM uses in https://github.com/llvm/llvm-project/blob/main/llvm/lib/Target/AArch64/AArch64.td, though rustc has some renaming logic for some feature names.

**Comment by jacobbramley** [comment]
Yes, I saw reference to that in a comment in rustc (in your first link).
I think there may be a missing translation there, actually: "fp16" is called "fphp" in Linux and the tests don't detect it (when other tools do). I've not followed the logic through the whole stack yet; consider it a strong hunch :-) Since finding hardware with _all_ of the features is difficult, I intend to test what I can, and audit the the whole list to check that there aren't similar issues elsewhere.

**Comment by Amanieu** [comment]
The feature detection code was fixed in #1005. However the submodule in rust-lang/rust hasn't been updated yet.
